jest/jest-tests.ts

```diff
@@ -367,3 +425,53 @@ describe('strictNullChecks', function () {
         done();
     })
 });
+
+class TestApi {
+    constructor() { };
+    testProp: boolean;
+    testMethod(a: number): string { return ""; }
+}
+
+declare function mockedFunc(a: number): string;
+
+describe('Mocked type', function () {
+    it('Works', function () {
+        const mock: jest.Mocked<TestApi> = new TestApi() as any;
+        mock.testProp;
```

Comment:
Yes, it's wrong, but there is no way to prevent it until TS will support some kind of conditional mapping. At least better than nothing i think